internal/api/metadata/describe.go

```diff
@@ -14,7 +17,35 @@ type DescribeResponse struct {
 
 // Describe all endpoints on the server.
 func HandleDescribe(request *DescribeRequest) (*DescribeResponse, *core.APIError) {
-	response := DescribeResponse{core.GetAPIDescription()}
+	response := DescribeResponse{}
+
+	if !request.ForceCompute {
+		apiDescription, err := core.GetAPIDescription()
+		if err != nil {
+			log.Warn("Unable to get cached API description.", err)
+		}
+
+		if apiDescription != nil {
+			response.APIDescription = *apiDescription
+			return &response, nil
+		}
+	}
+
+	routes := core.GetAPIRoutes()
+	if routes == nil || len(*routes) == 0 {
+		return nil, core.NewInternalError("-501", request, "Unable to describe API endpoints when the cached routes are empty.")
+	}
+
+	apiDescription, err := core.DescribeRoutes(*routes)
+	if err != nil {
+		return nil, core.NewInternalError("-502", request, "Failed to describe API endpoints.").Err(err)
+	}
+
+	core.SetAPIDescription(apiDescription)
```

Comment:
I'm not a fan of this call being here.
It seems a bit off that an endpoint is setting a cache inside of code
(it feels like core should be the one setting its own cache).

What if GetAPIDescription() accepted a boolean for forcing the recompute?
Then, most of this function would fall inside GetAPIDescription().

It's also fair to think about if you need this cache at all.
Aside from testing, it would just save one disk read.